Removed api-key and workshop functionality

ms-johnalex opened the pull request:
## Purpose
* Removed the api-key auth functionality from chat,py and related files to make the sample truly keyless
* Removed GitHub workshop functionality -- GitHub Model and Local Model support

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[X] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe:

* Using an api key has been removed from the sample
* Using a model from github models has been removed from the sample
* Using a local model with Ollama has been removed from the sample. 
``
```

## How to Test
*  Get the code
*  Run as directed in readme. 
```
git clone https://github.com/ms-johnalex/openai-chat-app-quickstart.git
cd openai-chat-app-quickstart
git checkout -b remove-api-key-auth-092624
```

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
```
```

## What to Check
Verify that the following are valid
* ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

pamelafox reviewed (commented):
FYI, lots of files showing up here with whitespace-only changes, can you do "git checkout main" for each of those, to reduce the size of this PR?

pamelafox commented on README.md:
```diff
@@ -1,216 +1,168 @@
-* [Getting started](#getting-started)
-  * [GitHub Codespaces](#github-codespaces)
-  * [VS Code Dev Containers](#vs-code-dev-containers)
-  * [Local Environment](#local-environment)
```
Why was local env removed? We usually support developers opening locally as well.

pamelafox commented on README.md:
````diff
@@ -1,216 +1,168 @@
+    azd env get-value AZURE_OPENAI_ENDPOINT
+    ```
+
+53. Start the development server:
````
You could remove "For use with Azure OpenAI" in step 2, since this will always use Azure OpenAI

pamelafox commented on docs/deploy_existing.md:
````diff
@@ -1,35 +1,32 @@
+azd env set AZURE_OPENAI_SKU_NAME {name of the SKU, defaults to "S0"}
+```
+
+If you don't want to deploy a new Azure OpenAI resource and just want to use an existing one via its endpoint, you can set the following values:
````
Can you add a note that an existing endpoint would need to have the appropriate RBAC role assigned?

ms-johnalex replied:
Yes.

ms-johnalex commented on .env.sample:
```diff
@@ -1,18 +1,7 @@
-# For local models, like Ollama/llamafile:
-LOCAL_MODELS_ENDPOINT="http://localhost:11434/v1"
-LOCAL_MODELS_NAME="llama3.1"
+OPENAI_HOST="azure"
```
No. I'll remove.

pamelafox commented on infra/main.parameters.json:
```diff
@@ -1,63 +1,60 @@
+        "value": "${AZURE_OPENAI_DEPLOYMENT_CAPACITY=30}"
+      },
+      "openAiDeploymentSkuName": {
+        "value": "${AZURE_OPENAI_DEPLOYMENT_SKU_NAME=GlobalStandard}"
```
I see this uses GlobalStandard now, are the locations accurate in main.bicep for the target model/version?

ms-johnalex replied:
Good catch. That's a testing artifact.  I'll set it back to Standard

pamelafox commented on src/quartapp/chat.py:
```diff
@@ -1,104 +1,80 @@
+
+@bp.before_app_serving
+async def configure_openai():
+    openai_host = os.getenv("OPENAI_HOST")
```
openai_host var no longer needed

pamelafox reviewed (changes requested):
See inline comments

pamelafox commented:
This also seems to have a merge conflict with main btw

ms-johnalex commented:
> This also seems to have a merge conflict with main btw

It's the README Files.

pamelafox commented on docs/README.md:
```diff
@@ -1,204 +1,203 @@
+* [Features](#features)
+* [Architecture diagram](#architecture-diagram)
+* [Opening the project](#opening-the-project)
+
```
```suggestion
```

pamelafox commented on docs/README.md:
```diff
@@ -1,204 +1,203 @@
+
+    * [Azure Developer CLI (azd)](https://aka.ms/install-azd)
+    * [Python 3.10+](https://www.python.org/downloads/)
+    * [Redis](https://redis.io/download)
```
```suggestion
```

pamelafox commented:
Can you run the pre-commit hook? Formatting failed

ms-johnalex commented:
> Can you run the pre-commit hook? Formatting failed

Black doesn't like something I changed in testapp.py. 

> Can you run the pre-commit hook? Formatting failed

Added NewLine to EOF in testapp.py. Black is happy now.
